docs/instrument-thanos.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Instrumenting Thanos
+type: docs
+menu: thanos
+---
+
+# Instrumenting Thanos
+We use the Prometheus [Go Client](https://github.com/prometheus/client_golang) library to instrument Thanos itself. Following are the metrics exposed by thanos:
+
+```bash
+TYPE                NAME                                                                 HELP
+GAUGE               thanos_delete_delay_seconds                                          Configured delete delay in seconds.
```

Comment:
Thanks for using `promlinter`. However, the problem here is:
Promlinter is just a static analysis tool, which means it cannot catch some metrics registered in the runtime. So it cannot catch all the metrics instrumented in the codebase. 

For example, it cannot catch the registry prefix.

Reply:
Should we manually add those?

Reply:
There is, please use the same flow as in other docs like here: https://github.com/thanos-io/thanos/blob/f4a5904b0c3be7565298cf257ad43ac1942021cc/docs/components/compact.md#L318

This points to some output in some file and is controlled by script here https://github.com/thanos-io/thanos/blob/7b45f70759071d8d980925cdd94da5e391bcb6bc/scripts/genflagdocs.sh#L59

In future we will switch to mdox which makes it much simpler https://github.com/bwplotka/mdox#code-generation (we are working on it with @saswatamcode )

Reply:
Even if I auto-generate this page some of the names of the metrics will be misleading. So, how should I move forward by using promlinter? 
Maybe we can write a new script that runs Thanos using the quickstart script and use /metrics endpoint to get all the exposed metrics and follow similar flags flow as Bartek mentioned. WDYT?

Reply:
Get metrics from `/metrics` cannot cover all the metrics as well.
I think the problem is caused by the registry prefix pattern. It is not linter friendly and makes things complex. I think the best way is to remove the prefixes and add prefixes as metric namespace.

Reply:
> Even if I auto-generate this page some of the names of the metrics will be misleading.

Let's discuss this, are we sure we cannot get prefixed metric or metric names there with promlinter? (: cc @yeya24

Reply:
Related discussion https://github.com/prometheus/client_golang/pull/890